File: examples/maven-spring-compile/src/test/java/community/flock/wirespec/example/maven/custom/app/todo/TodoTest.java
package community.flock.wirespec.example.maven.custom.app.todo;

import static org.junit.jupiter.api.Assertions.assertFalse;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class TodoTest {

    @Autowired
    private TodoService todoService;

    @Test
    void testTodo() {
        assertFalse(todoService.getAllTodos().isEmpty());
    }
}

File: examples/gradle-ktor/src/test/kotlin/community/flock/wirespec/example/gradle/app/todo/TodoTest.kt
package community.flock.wirespec.example.gradle.app.todo

import org.junit.Test
import kotlin.test.assertTrue

interface TestContext {
    val todoService: TodoService
}

class TodoTest {
    @Test
    fun testTodo() = testContext {
        assertTrue { todoService.getAllTodos().isNotEmpty() }
    }
}

private fun testContext(test: TestContext.() -> Unit) = object : TestContext {
    override val todoService = object :
        TodoService {
        override val todoRepository = LiveTodoRepository()
    }
}.test()
